pkg/uhttp/wrapper.go

```diff
@@ -148,8 +149,21 @@ func (c *BaseHttpClient) Do(req *http.Request, options ...DoOption) (*http.Respo
 		}
 	}
 
+	switch resp.StatusCode {
+	case http.StatusTooManyRequests:
+		return resp, status.Error(codes.Unavailable, resp.Status)
+	case http.StatusNotFound:
+		return resp, status.Error(codes.NotFound, resp.Status)
+	case http.StatusUnauthorized:
+		return resp, status.Error(codes.Unauthenticated, resp.Status)
+	case http.StatusForbidden:
+		return resp, status.Error(codes.PermissionDenied, resp.Status)
+	case http.StatusNotImplemented:
+		return resp, status.Error(codes.Unimplemented, resp.Status)
+	}
+
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		return resp, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
```

Comment:
ideally this would... use a baton-uthtp.Error type right? so you can add interface methods to get more info.  eg, the like the AWS Error type in their SDK?

Reply:
If we did that, wouldn't we have to check for baton uhttp errors in syncer.go's shouldWaitAndRetry()? And if we had some non-http error that was retryable, we'd have to check that in the same place. If, alternatively, we made general baton error types, they'd probably end up looking a lot like grpc's statuses, and I figured that grpc's statuses are better than whatever we'd make on our own. They correspond to most http statuses and have docs about what each error means: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md

Reply:
The big issue with gprc.Status objects is out of the box you loose the ability to keep more HTTP status around.... so its reallty that some APIs are weird.

Reply:
Alternatively, maybe the way a connector that needs "weird" handling is it could inject a HookError callback, and let it map it in a non-default way